src/nisarqa/utils/metadata_checks.py

```diff
@@ -220,143 +221,143 @@ def verify_metadata_cubes(
-    # Flag indicating if metadata datasets pass all verification checks; used
+    summary = nisarqa.get_summary()
+
+    # Very old test datasets do not necessarily have these calibration
+    # test datasets, and/or they are formatted differently.
+    # Since these are old and unsupported test datasets, ok to skip checks.
+    spec = nisarqa.Version.from_string(product.product_spec_version)
+    if spec < nisarqa.Version(1, 1, 0):
+        msg = (
+            "Input product has an unsupported product specification version."
+            " Verification checks of calibration information metadata skipped."
+        )
+        log.warning(msg)
+        summary.check_calibration_metadata(
+            result="WARN",
+            notes=msg,
+        )
+        return
+
+    # Flag indicating if metadata LUTs pass all verification checks; used
     # for Summary CSV reporting
     passes = True
     has_finite = True
 
-    # Check metadata datasets in metadata Group
-    try:
-        # Note: During the __post_init__ of constructing each metadata dataset,
-        # several validation checks are performed, including ensuring that
-        # there are corresponding datasets with x coordinates and y coordinates
-        # of the correct length. If these elements are missing, exceptions
-        # will get thrown.
-        spec = nisarqa.Version.from_string(product.product_spec_version)
-
-        for freq in product.freqs:
-
-            if spec < nisarqa.Version(1, 1, 0):
-                break
-
-            for ds in product.metadata_neb_datasets(freq):
-                has_finite &= _dataset_has_finite_pixels(ds)
-                passes &= has_finite
-                passes &= _dataset_has_finite_pixels(ds)
-                passes &= _dataset_is_not_all_zeros(ds)
-                passes &= _check_gdal(product=product, ds=ds)
-
-            for ds in product.metadata_elevation_antenna_pat_datasets(freq):
-                has_finite &= _dataset_has_finite_pixels(ds)
+    # Check calibration metadata LUTs in metadata Group
+    for freq in product.freqs:
+        calib_luts = chain(
+            product.metadata_neb_luts(freq),
+            product.metadata_elevation_antenna_pat_luts(freq),
+        )
+        if isinstance(product, nisarqa.SLC):
+            calib_luts = chain(calib_luts, product.metadata_geometry_luts())
+        if isinstance(product, nisarqa.RSLC):
+            calib_luts = chain(calib_luts, product.metadata_crosstalk_luts())
+
+        try:
+            # Note: During the __post_init__ of constructing each metadata LUT,
+            # several validation checks are performed, including ensuring that
+            # there are corresponding datasets with x coordinates and
+            # y coordinates of the correct length. If these elements are
+            # missing, exceptions will get thrown.
+            for ds in calib_luts:
+                has_finite &= _lut_has_finite_pixels(ds)
                 passes &= has_finite
```

Comment:
>`passes &= has_finite`

No need to do this at every iteration -- we could just do it once after the loop.